### DISCUSSION TITLE
A diff strategy

### DISCUSSION THREAD

**Discussion by isaksamsten** [olimorris/codecompanion.nvim#109] [Ideas]
Hi!

I've been using a plugin called [dante.nvim](https://github.com/S1M0N38/dante.nvim) for small repetitive tasks such as grammar checking. It has this great feature where you apply a prompt to a visual selection (or the current line) and then produces the changes in a split view with the context still intact around the changes and then uses the built in diff engine in Neovim to diff the two buffers.

**Example**

https://github.com/user-attachments/assets/88aecb3e-0f64-444c-ba99-fdb35a9a9c49

This strategy would be similar to that of "inline" but with more diff options (it's also easier to spot what has been changed)

Best,
Isak

**Comment by olimorris** [comment]
Yep, this is an awesome idea. I'll put it on the backlog (which currently is a list of todo's in Bear 😂).

Priority for next couple of days is to refactor messages in the chat buffer. Then I'll work on this.

Thanks for the awesome suggestion

**Reply by olimorris** [reply]
@isaksamsten before I embark on this, what's your feedback on the current diff implementation? And what tasks do you typically get CodeCompanion to do?

**Reply by isaksamsten** [reply]
I haven’t really used the inline strategy. I tried it out but usually I’m after changing smaller regions of the input. For instance, I use some prompts for language checking and rephrasing. For these used cases I’m not always interested in replacing the entire text but pick those changes that make sense. For code, some times it likes to replace if statements with expressions or the other way around. So for me accepting the prompt is usually not enough.

As of now, use Dante to check language and improve writing. I also have a prompt to “fix” code snippets which I rarely use. CodeCompanion I use to get suggestions on how to improve my code or discuss performance optimisation or stylistic choices. I’m also testing prompts for generating documentation strings.

I also like that I can use the muscle memory I’ve developed when diffing code when merging and transfer that to my AI assistant.

**Reply by SDGLBL** [reply]
@olimorris maybe https://github.com/yetone/avante.nvim `s diff implication is a better choice

**Comment by olimorris** [comment]
Progress so far:

![2024-08-28 22_47_05 - WezTerm](https://github.com/user-attachments/assets/e0e8928d-5bdd-4611-bf76-c24ffff1d43c)

I think the in-built diff is the way forward. So much less complexity and removes the reliance on the repo being git enabled.

**Reply by isaksamsten** [reply]
Wow! Amazing work! I agree that the built in diff engine is the way forward. It integrates so well!
